### DISCUSSION TITLE
The right way to deploy applications on teks with terragrunt

### DISCUSSION THREAD

**Discussion by tsu84** [particuleio/teks#124] [Q&A]
Hi,

I've managed to install teks on AWS and tried to deploy apps with kubectl and everything works.

Now I am migrating the app infrastructure to terragrunt, but I cannot figure out which is the right way to get the configurations from teks to deploy apps on it.

What I am trying to do is to import teks configuration as a (local) external module, but I am not sure if it is correct.

Can someone tell me how can I make deployments on multiple environments seamlessly with teks?

**Comment by ArchiFleKs** [comment]
Hi @tsu84 teks is a template for repository infrastructure, it is an opinionated example of what an infrastructure as code repository should look like.

It is meant to be cloned and adapted. If you want to build the module yourself you should look at the modules used in tEKS:
* [Terraform Kubernetes Addons](https://github.com/particuleio/terraform-kubernetes-addons/tree/main/modules/aws)
* [Official AWS Terraform modules](https://github.com/terraform-aws-modules)
